core/modules/l4_checksum.cc

```diff
@@ -49,27 +51,40 @@ void L4Checksum::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
     Ethernet *eth = batch->pkts()[i]->head_data<Ethernet *>();
 
     // Calculate checksum only for IPv4 packets
-    if (eth->ether_type != be16_t(Ethernet::Type::kIpv4))
+    if (eth->ether_type != be16_t(Ethernet::Type::kIpv4)) {
+      EmitPacket(ctx, batch->pkts()[i], FORWARD_GATE);
```

Comment:
To be safe, could you include a note in the protobuf about how non-IP packets are simply forwarded with no modification.